src/api/chrome.js

```diff
@@ -215,3 +218,32 @@ const getBadgeCounter = async (initialValue = 0) => {
   }
   return badgeCounter
 }
+
+/**
+ * Retrieves current tab url and validates it. Returns valid tab url otherwise undefined.
+ * @returns the tab url or undefined
+ */
+
+export const getCurrentTabUrlBackground = async () => {
+  let queryOptions = { active: true }
+  const currWindow = await chrome.windows.getCurrent()
+  // `tab` will either be a `tabs.Tab` instance or `undefined`.
+  let [tab] = await chrome.tabs.query({
+    ...queryOptions,
+    windowId: currWindow.id
+  })
+
+  if (!isValidUrl(tab.url, URL_MATCH)) {
+    return undefined
+  }
+  return tab.url
```

Comment:
returning undefined from function is not js idiomatic. invert the condition and simply don't return anything if the url does not match

Reply:
I agree that inverting condition is a good idea but I'd rather return undefined explicitly at the end of the function(only for readability purpose). wdyt?